src/rules/max-line-length/index.js

```diff
@@ -50,7 +50,8 @@ export default function (maxLength, options) {
     }
 
     function checkNewline(match) {
-      let nextNewlineIndex = rootString.indexOf("\n", match.endIndex)
+      const endOfLineIndicator = (rootString.indexOf("\r\n", match.endIndex) !== -1) ? "\r\n" : "\n"
```

Comment:
@JasonAMartin one more note here: I don't think this is the most efficient way to check for the difference — searching the entire root string for every newline. What we've done in some other places to accommodate column placement for `\r\n` is just to adjust the position right before reporting the warning. We can find `\n` and do all the analysis based on `\n`; then just before reporting check if there's an `\r` in front of it and adjust the index if so. Does that sound like it would work here? Would certainly avoid a lot of needless string searching.